--- a/net/minecraft/src/BlockFlower.java
+++ b/net/minecraft/src/BlockFlower.java
@@ -19,8 +19,8 @@
 		return id == Block.grass.blockID || id == Block.dirt.blockID || id == Block.tilledField.blockID;
 	}
 
-	public void onNeighborBlockChange(World world, int x, int y, int z, int flag) {
-		super.onNeighborBlockChange(world, x, y, z, flag);
+	public void doPhysics(World world, int x, int y, int z, int flag) {
+		super.doPhysics(world, x, y, z, flag);
 		this.dropIfNeighborPlaced(world, x, y, z);
 	}
 
@@ -30,7 +30,7 @@
 
 	protected final void dropIfNeighborPlaced(World world, int x, int y, int z) {
 		if(!this.canBlockStay(world, x, y, z)) {
-			this.dropBlockAsItem(world, x, y, z, world.getBlockMetadata(x, y, z));
+			this.dropNaturally(world, x, y, z, world.getBlockMetadata(x, y, z));
 			world.setBlockWithNotify(x, y, z, 0);
 		}
 
