--- a/net/minecraft/src/BlockDoor.java
+++ b/net/minecraft/src/BlockDoor.java
@@ -112,15 +112,15 @@
 		}
 	}
 
-	public void onNeighborBlockChange(World world, int x, int y, int z, int flag) {
+	public void doPhysics(World world, int x, int y, int z, int flag) {
 		int i6 = world.getBlockMetadata(x, y, z);
 		if((i6 & 8) != 0) {
 			if(world.getBlockId(x, y - 1, z) != this.blockID) {
 				world.setBlockWithNotify(x, y, z, 0);
 			}
 
-			if(flag > 0 && Block.canBlockGrass[flag].canProvidePower()) {
-				this.onNeighborBlockChange(world, x, y - 1, z, flag);
+			if(flag > 0 && Block.blockList[flag].canProvidePower()) {
+				this.doPhysics(world, x, y - 1, z, flag);
 			}
 		} else {
 			boolean z7 = false;
@@ -138,8 +138,8 @@
 			}
 
 			if(z7) {
-				this.dropBlockAsItem(world, x, y, z, i6);
-			} else if(flag > 0 && Block.canBlockGrass[flag].canProvidePower()) {
+				this.dropNaturally(world, x, y, z, i6);
+			} else if(flag > 0 && Block.blockList[flag].canProvidePower()) {
 				boolean z8 = world.isBlockIndirectlyGettingPowered(x, y, z) || world.isBlockIndirectlyGettingPowered(x, y + 1, z);
 				this.onPoweredBlockChange(world, x, y, z, z8);
 			}
