--- a/net/minecraft/src/BlockCrops.java
+++ b/net/minecraft/src/BlockCrops.java
@@ -84,8 +84,8 @@
 				float f9 = world.rand.nextFloat() * f7 + (1.0F - f7) * 0.5F;
 				float f10 = world.rand.nextFloat() * f7 + (1.0F - f7) * 0.5F;
 				EntityItem entityItem11 = new EntityItem(world, (double)((float)x + f8), (double)((float)y + f9), (double)((float)z + f10), new ItemStack(Item.seeds));
-				entityItem11.delayBeforeCanPickup = 10;
-				world.spawnEntityInWorld(entityItem11);
+				entityItem11.pickupDelay = 10;
+				world.addEntity(entityItem11);
 			}
 		}
 
