--- a/net/minecraft/src/BlockStationary.java
+++ b/net/minecraft/src/BlockStationary.java
@@ -12,8 +12,8 @@
 
 	}
 
-	public void onNeighborBlockChange(World world, int x, int y, int z, int flag) {
-		super.onNeighborBlockChange(world, x, y, z, flag);
+	public void doPhysics(World world, int x, int y, int z, int flag) {
+		super.doPhysics(world, x, y, z, flag);
 		if(world.getBlockId(x, y, z) == this.blockID) {
 			this.setNotStationary(world, x, y, z);
 		}
@@ -43,7 +43,7 @@
 						world1.setBlockWithNotify(i2, i3, i4, Block.fire.blockID);
 						return;
 					}
-				} else if(Block.canBlockGrass[i8].material.getIsSolid()) {
+				} else if(Block.blockList[i8].material.getIsSolid()) {
 					return;
 				}
 			}
