--- a/net/minecraft/src/BlockStairs.java
+++ b/net/minecraft/src/BlockStairs.java
@@ -53,7 +53,7 @@
 		this.setBlockBounds(0.0F, 0.0F, 0.0F, 1.0F, 1.0F, 1.0F);
 	}
 
-	public void onNeighborBlockChange(World world, int x, int y, int z, int flag) {
+	public void doPhysics(World world, int x, int y, int z, int flag) {
 		if(world.getBlockMaterial(x, y + 1, z).isSolid()) {
 			world.setBlockWithNotify(x, y, z, this.modelBlock.blockID);
 		} else {
@@ -68,7 +68,7 @@
 			this.g(world, x, y + 1, z + 1);
 		}
 
-		this.modelBlock.onNeighborBlockChange(world, x, y, z, flag);
+		this.modelBlock.doPhysics(world, x, y, z, flag);
 	}
 
 	private void g(World world, int x, int y, int z) {
@@ -139,7 +139,7 @@
 
 	private boolean isBlockStair(World world, int x, int y, int z) {
 		int i5 = world.getBlockId(x, y, z);
-		return i5 == 0 ? false : Block.canBlockGrass[i5].getRenderType() == 10;
+		return i5 == 0 ? false : Block.blockList[i5].getRenderType() == 10;
 	}
 
 	public void onBlockClicked(World world1, int i2, int i3, int i4, EntityPlayer entityPlayer5) {
@@ -187,7 +187,7 @@
 	}
 
 	public void onBlockAdded(World world1, int i2, int i3, int i4) {
-		this.onNeighborBlockChange(world1, i2, i3, i4, 0);
+		this.doPhysics(world1, i2, i3, i4, 0);
 		this.modelBlock.onBlockAdded(world1, i2, i3, i4);
 	}
 
@@ -199,8 +199,8 @@
 		this.modelBlock.dropBlockAsItemWithChance(world, x, y, z, i5, f6);
 	}
 
-	public void dropBlockAsItem(World world1, int i2, int i3, int i4, int i5) {
-		this.modelBlock.dropBlockAsItem(world1, i2, i3, i4, i5);
+	public void dropNaturally(World world1, int i2, int i3, int i4, int i5) {
+		this.modelBlock.dropNaturally(world1, i2, i3, i4, i5);
 	}
 
 	public void onEntityWalking(World world1, int i2, int i3, int i4, Entity entity5) {
