--- a/net/minecraft/src/BlockSnowBlock.java
+++ b/net/minecraft/src/BlockSnowBlock.java
@@ -18,7 +18,7 @@
 
 	public void updateTick(World world1, int i2, int i3, int i4, Random random5) {
 		if(world1.getSavedLightValue(EnumSkyBlock.Block, i2, i3, i4) > 11) {
-			this.dropBlockAsItem(world1, i2, i3, i4, world1.getBlockMetadata(i2, i3, i4));
+			this.dropNaturally(world1, i2, i3, i4, world1.getBlockMetadata(i2, i3, i4));
 			world1.setBlockWithNotify(i2, i3, i4, 0);
 		}
 
